CHANGELOG.md

```diff
@@ -11,6 +11,7 @@
 * Added RealmQuery.distinct() and RealmResults.distinct() (#1568).
 * Fixed bug when multiple calls of RealmResults.distinct() causes to return wrong results (#2198).
 * RealmResults.size() now returns Integer.MAX_VALUE when actual size is greater than Integer.MAX_VALUE (#2129).
+* Added RealmQuery.distinctAsync() and RealmResults.distinctAsync() (#2118).
```

Comment:
We need to add a BREAKING CHANGE section, outlining that we now throw IllegalArgumentException instead of UnsupportedOperationException.